sdks/cpp/lite/include/ParamWithValue.h

```diff
@@ -70,17 +72,40 @@ class ParamWithValue : public catena::common::IParam {
     virtual ~ParamWithValue() = default;
 
     void toProto(catena::Value& value) const override {
-        
+        catena::lite::toProto<T>(value, &value_.get());
+    }
+
+    void toProto(catena::Value& value, void* src) const override {
+        catena::lite::toProto<T>(value, src);
     }
 
     void fromProto(catena::Value& value) override {
-        
+        catena::lite::fromProto<T>(&value_.get(), value);
+        catena::lite::fromProto<T>(&value_.get(), value);
     }
 
+    /**
+     * @brief deserialize the parameter value from protobuf
+     * @param dst the destination value
+     * @param value the protobuf value to deserialize from
+     * @note should only be called for non structured types
+     */
+    void fromProto(void* dst, catena::Value& value) override {
+        assert(dst == &value_.get());
```

Comment:
Why is this here? If dst and value.get() are the same then why not remove the dst parameter?